src/Principal.mo

```diff
@@ -1,58 +1,241 @@
-
-  /// Returns `x <= y`.
-  public func lessOrEqual(x : Principal, y : Principal) : Bool { x <= y };
-
-  /// Returns `x > y`.
-  public func greater(x : Principal, y : Principal) : Bool { x > y };
-
-  /// Returns `x >= y`.
-  public func greaterOrEqual(x : Principal, y : Principal) : Bool { x >= y };
-
-  /// Returns the order of `x` and `y`.
-  public func compare(x : Principal, y : Principal) : {
+  /// General purpose comparison function for `Principal`. Returns the `Order` (
+  /// either `#less`, `#equal`, or `#greater`) of comparing `principal1` with
+  /// `principal2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let principal1 = Principal.fromText("un4fu-tqaaa-aaaab-qadjq-cai");
+  /// let principal2 = Principal.fromText("un4fu-tqaaa-aaaab-qadjq-cai");
+  /// Principal.compare(principal1, principal2) // => #equal
+  /// ```
+  public func compare(principal1 : Principal, principal2 : Principal) : {
     #less;
     #equal;
     #greater
   } {
-    if (x < y) { #less } else if (x == y) { #equal } else { #greater }
+    if (principal1 < principal2) {
+      #less
+    } else if (principal1 == principal2) {
+      #equal
+    } else {
+      #greater
+    }
+  };
+
+  /// Equality function for Principal types.
+  /// This is equivalent to `principal1 == principal2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let principal1 = Principal.fromText("un4fu-tqaaa-aaaab-qadjq-cai");
+  /// let principal2 = Principal.fromText("un4fu-tqaaa-aaaab-qadjq-cai");
+  /// ignore Principal.equal(principal1, principal2);
+  /// principal1 == principal2 // => true
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `==` operator) is so that you can use it as a higher order
+  /// function. It is not possible to use `==` as a higher order function at the moment.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// import Buffer "mo:base/Buffer";
+  ///
+  /// let buffer1 = Buffer.Buffer<Principal>(3);
+  /// let buffer2 = Buffer.Buffer<Principal>(3);
+  /// Buffer.equal(buffer1, buffer2, Principal.equal) // => true
+  /// ```
+  public func equal(principal1 : Principal, principal2 : Principal) : Bool {
+    principal1 == principal2
+  };
+
+  /// Inequality function for Principal types.
+  /// This is equivalent to `principal1 != principal2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let principal1 = Principal.fromText("un4fu-tqaaa-aaaab-qadjq-cai");
+  /// let principal2 = Principal.fromText("un4fu-tqaaa-aaaab-qadjq-cai");
+  /// ignore Principal.notEqual(principal1, principal2);
+  /// principal1 != principal2 // => false
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `!=` operator) is so that you can use it as a higher order
+  /// function. It is not possible to use `!=` as a higher order function at the moment.
```

Comment:
```suggestion
  /// to the existing `!=` operator) is so that you can use it as a function value.
  /// It is not possible to use `!=` as a function value at the moment.
```